plugin/session_buffer.py

```diff
@@ -83,6 +92,13 @@ def __init__(self, session_view: SessionViewProtocol, buffer_id: int, uri: Docum
         self.total_warnings = 0
         self.should_show_diagnostics_panel = False
         self.diagnostics_debouncer = Debouncer()
+        self.semantic_token_types = None  # type: Optional[List[str]]
+        self.semantic_token_modifiers = None  # type: Optional[List[str]]
+        self.semantic_tokens_pending_request = False
+        self.semantic_tokens_data = []  # type: List[int]
+        self.semantic_tokens_result_id = None  # type: Optional[str]
+        self.semantic_tokens = []  # type: List[SemanticToken]
+        self.semantic_tokens_keys = set()  # type: Set[str]
```

Comment:
Perhaps we can separate this out into a class called `SemanticTokenHandler` or something? There's a lot of repetitious identifier names here.